pkg/controllers/allocation/controller.go

```diff
@@ -133,6 +141,20 @@ func (c *Controller) Register(ctx context.Context, m manager.Manager) error {
 		Watches(
 			&source.Kind{Type: &v1.Pod{}},
 			handler.EnqueueRequestsFromMapFunc(c.podToProvisioner),
+			// Only process pod update events
+			builder.WithPredicates(
+				predicate.Funcs{
+					CreateFunc: func(_ event.CreateEvent) bool {
```

Comment:
Consider inlining to make this smaller

`CreateFunc: func(_ event.CreateEvent) bool { return false }`